training/MESSAGING_OVERVIEW.md

```diff
@@ -57,21 +57,19 @@ The three components of the `registerFunction` method are the `cmd` string, the
 
 It is important the Chapel function is formatted correctly or the server will not compile. The expected formatting for a `serverFunctionMsg` is:
 ```chapel
-proc serverFunctionMsg(cmd: string, payload: string, argSize: int, st: borrowed SymTab): MsgTuple throws {}
+proc serverFunctionMsg(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab): MsgTuple throws {}
 ```
 
 The parameters in this function each server a specific purpose:
 - `cmd` is the command string used to access this function. It is used mostly for debugging as the `cmd` string used to access the function will be captured in the logger messages.
-- `payload` is the JSON string of arguments passed from the client.
-- `argSize` lets Chapel know how many arguments are in the payload JSON so it can properly create a map using the JSON.
+- `msgArgs` is the recieved information that the client passes as a `repMsg`. The information about the number of arguements ia bundled and will be pared accordingly with the denoted passed types. Individual arguements will need to be extracted on the server side.
 - `st: borrowed SymTab` is a reference parameter to the Server's Symbol Table. The Symbol Table acts like Arkouda's database; storing all information related to a data structure that can then be referenced from the Client without having to pass large amounts of data back and forth.
 
-All messages sent to the server are in JSON format. Before any arguments sent from the client can be used, they must be parsed. To do this use the function `parseMessageArgs(payload, argSize)`. This will parse your message arguments into a Map that can then be accessed using accessor methods. The most common of these accessor methods is `getValueOf(argName)` which will return the value of an argument that matches the provided name.
+All messages sent to the server are in JSON format. Before any arguments sent from the client can be used, they must be parsed. Each message arguement that is sent to the server needs to be parsed into its own variable to be accessed, using a Map that can then be accessed using accessor methods. The most common of these accessor methods is `getValueOf(argName)` which will return the value of an argument that matches the provided name.

```

Comment:
The parsing now happens automatically, so I am not sure we want to indicate this to the user. I would state that the JSON sent to the server is used to generate a MessageArgs object. That object maps the parameter name to the value. The values are always strings, but functions are provided to access it as the correct type for processing.